easybuild/tools/toolchain/toolchain.py

```diff
@@ -688,8 +693,18 @@ def prepare(self, onlymod=None, silent=False, loadmod=True, rpath_filter_dirs=No
         :param rpath_filter_dirs: extra directories to include in RPATH filter (e.g. build dir, tmpdir, ...)
         :param rpath_include_dirs: extra directories to include in RPATH
         """
+
+        # do all dependencies have a toolchain version?
+        if deps is None:
+            deps = []
+        tcdeps = self._check_dependencies(deps)
```

Comment:
@bartoldeman Why not make this `self.dependencies = self._check_dependencies(deps)`?

Then there's no need to change `_load_modules`, `_load_dependencies_modules` and `_add_dependency_variables` to pass down `tcdeps`, they can just keep using `self.dependencies` like they were before?

It's a bit silly to pass them around if we have `self.dependencies` anyway (which we can't remove since easyblocks may be using it)...